src/query/functions/temporal/aggregation.go

```diff
@@ -85,28 +92,42 @@ func NewAggOp(args []interface{}, optype string) (transform.Params, error) {
 			aggFunc: aggregationFunc,
 		}
 
+		if optype == QuantileType {
+			if len(args) != 2 {
+				return emptyOp, fmt.Errorf("invalid number of args for %s: %d", QuantileType, len(args))
+			}
+
+			scalar, ok := args[0].(float64)
+			if !ok {
+				return emptyOp, fmt.Errorf("unable to cast to scalar argument: %v for %s", args[1], QuantileType)
+			}
+
+			a.quantileScalar = scalar
+		}
+
 		return newBaseOp(args, optype, a)
 	}
 
 	return nil, fmt.Errorf("unknown aggregation type: %s", optype)
 }
 
 type aggNode struct {
-	op         baseOp
-	controller *transform.Controller
-	aggFunc    func([]float64) float64
+	op             baseOp
+	controller     *transform.Controller
+	aggFunc        func([]float64, float64) float64
+	quantileScalar float64
 }
 
 func (a *aggNode) Process(datapoints ts.Datapoints, _ time.Time) float64 {
-	return a.aggFunc(datapoints.Values())
+	return a.aggFunc(datapoints.Values(), a.quantileScalar)
 }
 
-func avgOverTime(values []float64) float64 {
+func avgOverTime(values []float64, _ float64) float64 {
```

Comment:
Not great that we modify all the func signatures for a var that isn't used in most of these functions. If it needs to be this way, can we wrap it in an options or something at least so the function signature doesn't have to ever change again? (But only if you have to pass this all the way down, see comment above).

Reply:
Or consider putting this in some sort of context object that you propagate instead of a specific options for agg functions.

Reply:
+1. Would it make sense to pass in `scalar` as a constructor argument to the quantile `aggFunc`? That way we separate out the "configuration" of the `aggFunc` from it's actual execution. A note on how that could work: 

1. We can switch `aggFunc` to be an interface (functions can implement interfaces, so this is actually less work than it seems), to allow for `quantile` to hold configuration (`scalar`)
2. Pass `scalar` to `quantile` when we initialize the node instead (threading through `NewAggOp` and `temporal.baseOp.Node`)